### PR TITLE
Add typescript

### PR DESCRIPTION
**Pull request by Baksha2008**

### CODE REVIEW

File: src/components/AddButton.tsx [hunk 1]
```diff
@@ -1,9 +1,13 @@
 import React from 'react';
 import { TouchableHighlight, StyleSheet } from 'react-native';
 import { FontAwesome } from '@expo/vector-icons';
-import { COLORS } from 'constants/theme';
+import { COLORS } from '../constants/theme';
 
-export const AddButton = (props) => (
+interface IAddButtonProps {
+  onPress(): void;
```
Comment (rumbaa01) [inline src/components/AddButton.tsx]: Let's use ```onPress() => void``` instead

File: src/components/Notes.tsx [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
 import { View, Text, StyleSheet } from 'react-native';
 
-import { FONT_SIZE } from 'constants/theme';
-import { strings } from 'constants/strings';
+import { FONT_SIZE } from '../constants/theme';
+import { strings } from '../constants/strings';
 
-export class Notes extends Component {
+interface INotesProps {
+  notes: string;
```
Comment (rumbaa01) [inline src/components/Notes.tsx]: notes can be Object, Let's use ```notes: string | {[key: string]: string}```

File: src/components/TimeOptions/TimeOptions.tsx [hunk 3]
```diff
@@ -6,68 +6,77 @@ import { FontAwesome } from '@expo/vector-icons';
...
+  selectedHours: string;
+  selectedMinutes: string;
+  [key: string]: string;
+}
+interface IPropsTimeOption {
+  isValid: boolean;
+  option: string;
+  onDeleteTime?(index: number): void;
+  onAddTime?: (time: string) => void;
+  timeOptions?: any[];
```
Comment (rumbaa01) [inline src/components/TimeOptions/TimeOptions.tsx]: avoid any

File: src/components/TimeOptions/components/TimeInput.tsx [hunk 4]
```diff
@@ -1,8 +1,15 @@
...
-export class TimeInput extends Component {
-  onTimeChange = (value) => {
+interface IPropsTimeInput {
+  option: string;
+  onTimeChange(value: string, option: string): void;
+  value: string;
+  onAddTime?(): void;
+}
+
+export class TimeInput extends Component<IPropsTimeInput, {}> {
```
Comment (rumbaa01) [inline src/components/TimeOptions/components/TimeInput.tsx]: Don't pass empty object for state: ``` <IPropsTimeInput>```

File: src/components/TimeOptions/components/TimeItem.tsx [hunk 5]
```diff
@@ -1,19 +1,26 @@
...
+import { COLORS } from '../../../constants/theme';
+
+interface IPropsTimeItem {
+  onDeleteTime(index: number): void;
+  index: number;
+  time: string;
+}
 
-export class TimeItem extends PureComponent {
+export class TimeItem extends PureComponent<IPropsTimeItem, {}> {
```
Comment (rumbaa01) [inline src/components/TimeOptions/components/TimeItem.tsx]: Same

File: src/redux/actions/pillsAction.ts [hunk 6]
```diff
@@ -1,14 +1,14 @@
 export const ADD_PILL = 'ADD_PILL';
 export const TAKE_PILL = 'TAKE_PILL';
 
-export const addPill = (pill) => {
+export const addPill = (pill: any) => {
```
Comment (rumbaa01) [inline src/redux/actions/pillsAction.ts]: avoid any

File: src/redux/actions/pillsAction.ts [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
   return {
     type: ADD_PILL,
     payload: pill,
   };
 };
 
-export const onTakePill = (id, time) => {
+export const onTakePill = (id: number, time: any) => {
```
Comment (rumbaa01) [inline src/redux/actions/pillsAction.ts]: avoid any

File: src/redux/reducers/auth.ts [hunk 8]
```diff
@@ -1,6 +1,14 @@
-import { createReducer } from 'services/utils';
+import { createReducer } from '../../services/utils';
 import { LOG_IN, LOG_OUT } from '../actions/authActions';
 
+interface state {
+  ilLobin: boolean;
```
Comment (rumbaa01) [inline src/redux/reducers/auth.ts]: ilLobin ? typo ?

File: src/redux/reducers/auth.ts [hunk 9]
```diff
@@ -10,11 +18,11 @@ const INITIAL_STATE = {
 };
 
 const handlers = {
-  [LOG_IN]: (state, action) => {
+  [LOG_IN]: (state: state, action: any) => {
     const { name, picture } = action.payload;
     return { ...state, isLogin: true, user: { name, picture } };
   },
-  [LOG_OUT]: (state) => {
+  [LOG_OUT]: (state: state) => {
```
Comment (rumbaa01) [inline src/redux/reducers/auth.ts]: Let's use uppercase

File: src/redux/reducers/auth.ts [hunk 10]
```diff
@@ -10,11 +18,11 @@ const INITIAL_STATE = {
 };
 
 const handlers = {
-  [LOG_IN]: (state, action) => {
+  [LOG_IN]: (state: state, action: any) => {
```
Comment (rumbaa01) [inline src/redux/reducers/auth.ts]: avoid any

File: src/redux/reducers/eventReducer.ts [hunk 11]
```diff
@@ -1,18 +1,22 @@
-import { createReducer } from 'services/utils';
+import { createReducer } from '../../services/utils';
 import { ADD_EVENT, DELETE_EVENT } from '../actions/eventAction';
 
 const INITIAL_STATE = {
   events: [],
 };
 
+interface state {
+  events: any[];
```
Comment (rumbaa01) [inline src/redux/reducers/eventReducer.ts]: avoid any

File: src/redux/reducers/eventReducer.ts [hunk 12]
```diff
@@ -1,18 +1,22 @@
...
+}
+
 const handlers = {
-  [ADD_EVENT]: (state, action) => {
+  [ADD_EVENT]: (state: state, action: any) => {
```
Comment (rumbaa01) [inline src/redux/reducers/eventReducer.ts]: same

File: src/redux/reducers/eventReducer.ts [hunk 13]
```diff
@@ -1,18 +1,22 @@
...
     return {
       ...state,
       events: [...state.events, action.payload],
     };
   },
-  [DELETE_EVENT]: (state, action) => {
+  [DELETE_EVENT]: (state: state, action: any) => {
```
Comment (rumbaa01) [inline src/redux/reducers/eventReducer.ts]: same

File: src/redux/reducers/pillsReducer.ts [hunk 14]
```diff
@@ -1,23 +1,27 @@
...
   pills: [],
 };
 
+interface state {
+  pills: any[];
+}
+
 const handlers = {
-  [ADD_PILL]: (state, action) => {
+  [ADD_PILL]: (state: state, action: any) => {
```
Comment (rumbaa01) [inline src/redux/reducers/pillsReducer.ts]: same

File: src/redux/reducers/pillsReducer.ts [hunk 15]
```diff
@@ -1,23 +1,27 @@
...
       pills: [...state.pills, action.payload],
     };
   },
-  [TAKE_PILL]: (state, action) => {
+  [TAKE_PILL]: (state: state, action: any) => {
     const { pills } = state;
     pills.forEach((pill) => {
       if (pill.id === action.payload.id) {
-        pill.timeOptions.forEach((timeOption) => {
+        pill.timeOptions.forEach((timeOption: any) => {
```
Comment (rumbaa01) [inline src/redux/reducers/pillsReducer.ts]: same

File: src/screens/Auth/Auth.container.ts [hunk 16]
```diff
@@ -1,12 +1,12 @@
...
-import { logIn } from 'redux/actions/authActions';
+import { logIn } from '../../redux/actions/authActions';
 import { AuthComponent } from './Auth';
 
 export const Auth = connect(
   null,
   (dispatch) => {
     return {
-      logIn: (userData) => {
+      logIn: (userData: any) => {
```
Comment (rumbaa01) [inline src/screens/Auth/Auth.container.ts]: same

File: src/screens/Auth/Auth.tsx [hunk 17]
```diff
@@ -3,10 +3,25 @@ import { View } from 'react-native';
...
+import { authService } from '../../services/authService';
+import { HOME } from '../../constants/screenCases';
 
-export class AuthComponent extends React.Component {
+interface IAuthComponentState {
+  signedIn: boolean;
+  userData: any;
+}
+interface INavigation {
+  navigate(HOME: string): void;
```
Comment (rumbaa01) [inline src/screens/Auth/Auth.tsx]: Let's use ```route: string``` instead of HOME

File: src/screens/Auth/Auth.tsx [hunk 18]
```diff
@@ -3,10 +3,25 @@ import { View } from 'react-native';
...
 
-import { authService } from 'services/authService';
-import { HOME } from 'constants/screenCases';
+import { authService } from '../../services/authService';
+import { HOME } from '../../constants/screenCases';
 
-export class AuthComponent extends React.Component {
+interface IAuthComponentState {
+  signedIn: boolean;
+  userData: any;
```
Comment (rumbaa01) [inline src/screens/Auth/Auth.tsx]: any

File: src/screens/Auth/Auth.tsx [hunk 19]
```diff
@@ -3,10 +3,25 @@ import { View } from 'react-native';
...
+}
+interface INavigation {
+  navigate(HOME: string): void;
+}
+interface IAuthComponentProps {
+  logIn(obj: any): void;
```
Comment (rumbaa01) [inline src/screens/Auth/Auth.tsx]: What is obj? 
1. Change name
2. Avoid any

File: src/screens/Events/scenes/Events/Events.container.ts [hunk 20]
```diff
@@ -2,15 +2,15 @@ import { connect } from 'react-redux';
 
 import { EventsComponent } from './Events';
 
-import { deleteEvent } from 'redux/actions/eventAction';
+import { deleteEvent } from '../../../../redux/actions/eventAction';
 
-const mapStateToprops = (state) => ({
+const mapStateToprops = (state: any) => ({
```
Comment (rumbaa01) [inline src/screens/Events/scenes/Events/Events.container.ts]: any? I think, we have ```State``` type

File: src/screens/Events/scenes/Events/Events.tsx [hunk 21]
```diff
@@ -1,15 +1,34 @@
...
+import { COLORS, FONT_SIZE } from '../../../../constants/theme';
+import { ADD_EVENT } from '../../../../constants/screenCases';
+import { strings } from '../../../../constants/strings';
 
-export class EventsComponent extends Component {
-  renderPill = (item) => {
+import { IEvent } from '../EditEvent/EditEvent';
+
+interface navigation {
+  navigate(ADD_EVENT: string): void;
```
Comment (rumbaa01) [inline src/screens/Events/scenes/Events/Events.tsx]: same

File: src/screens/Events/scenes/Events/Events.tsx [hunk 22]
```diff
@@ -1,15 +1,34 @@
...
+
+interface IEventsComponent {
+  onAddEvent(): void;
+  onDeleteEvent(i: number): void;
+}
+
+export class EventsComponent extends Component<IEventsComponentProps, {}>
+  implements IEventsComponent {
+  renderPill = (item: any) => {
+    console.log(item);
```
Comment (rumbaa01) [inline src/screens/Events/scenes/Events/Events.tsx]: clean up pls

File: src/screens/Pills/scenes/AddPill/AddPill.tsx [hunk 23]
```diff
@@ -50,41 +91,41 @@ export class AddPillComponent extends Component {
     isDoseValid: false,
   };
 
-  onTitleChange = (title) => {
+  public onTitleChange = (title: string) => {
```
Comment (rumbaa01) [inline src/screens/Pills/scenes/AddPill/AddPill.tsx]: Why public? Let's use private instead

File: src/screens/Pills/scenes/AddPill/AddPill.tsx [hunk 24]
```diff
@@ -50,41 +91,41 @@ export class AddPillComponent extends Component {
...
           title,
         },
       }),
-      this.validate('isTitleValid', title)
+      () => this.validate('isTitleValid', title)
     );
   };
 
-  onAddPillType = (type) => {
+  public onAddPillType = (type: string) => {
```
Comment (rumbaa01) [inline src/screens/Pills/scenes/AddPill/AddPill.tsx]: same

File: src/screens/Pills/scenes/Pills/Pills.tsx [hunk 25]
```diff
@@ -2,18 +2,28 @@ import React from 'react';
...
+import { FONT_SIZE, COLORS } from '../../../../constants/theme';
+import { strings } from '../../../../constants/strings';
+import { ADD_PILL } from '../../../../constants/screenCases';
 
-export class PillsComponent extends React.PureComponent {
-  keyExtractor = (item) => {
+import { IPill } from '../AddPill/AddPill';
+
+interface INavigation {
+  navigate(ADD_PILL: string): void;
```
Comment (rumbaa01) [inline src/screens/Pills/scenes/Pills/Pills.tsx]: same

File: src/screens/Pills/scenes/Pills/Pills.tsx [hunk 26]
```diff
@@ -2,18 +2,28 @@ import React from 'react';
...
+}
+
+export class PillsComponent extends React.PureComponent<IPillsComponentProps> {
+  keyExtractor = (item: any) => {
     return item.id;
   };
 
-  renderPill = (item) => {
+  renderPill = (item: any) => {
     return <PillItem pill={item} key={item.id} />;
```
Comment (rumbaa01) [inline src/screens/Pills/scenes/Pills/Pills.tsx]: any

File: src/screens/Pills/scenes/Pills/components/PillItem/PillItem.tsx [hunk 27]
```diff
@@ -1,12 +1,16 @@
...
-import { Notes } from 'components/Notes';
+import { Notes } from '../../../../../../components/Notes';
 import { Checkboxes } from './components/Checkboxes';
-import { COLORS } from 'constants/theme';
+import { COLORS } from '../../../../../../constants/theme';
 import { Dose } from './components/Dose';
 
-export class PillItem extends Component {
+interface IPropsPillItem {
+  pill: any;
```
Comment (rumbaa01) [inline src/screens/Pills/scenes/Pills/components/PillItem/PillItem.tsx]: any

File: src/redux/reducers/auth.ts (hunk 8 above)
Comment (dimagreat) [inline src/redux/reducers/auth.ts]: iLobin?

File: src/redux/reducers/auth.ts [hunk 28]
```diff
@@ -1,6 +1,14 @@
-import { createReducer } from 'services/utils';
+import { createReducer } from '../../services/utils';
 import { LOG_IN, LOG_OUT } from '../actions/authActions';
 
+interface state {
```
Comment (dimagreat) [inline src/redux/reducers/auth.ts]: Interface should start from `I`

File: src/components/TimeOptions/components/TimeItem.tsx [hunk 29]
```diff
@@ -1,19 +1,26 @@
 import React, { PureComponent } from 'react';
 import { StyleSheet, Text, View, TouchableHighlight } from 'react-native';
+
 import { FontAwesome } from '@expo/vector-icons';
-import { COLORS } from 'constants/theme';
+import { COLORS } from '../../../constants/theme';
+
+interface IPropsTimeItem {
```
Comment (dimagreat) [inline src/components/TimeOptions/components/TimeItem.tsx]: `ITimeItemProps`

File: src/redux/reducers/pillsReducer.ts [hunk 30]
```diff
@@ -1,23 +1,27 @@
 import { ADD_PILL, TAKE_PILL } from '../actions/pillsAction';
 
-import { createReducer } from 'services/utils';
+import { createReducer } from '../../services/utils';
 
 const INITIAL_STATE = {
   pills: [],
 };
 
+interface state {
```
Comment (dimagreat) [inline src/redux/reducers/pillsReducer.ts]: from `I`

File: src/screens/Auth/Auth.tsx (hunk 19 above)
Comment (dimagreat) [inline src/screens/Auth/Auth.tsx]: obj is bad name

File: src/screens/Auth/Auth.tsx (hunk 17 above)
Comment (dimagreat) [inline src/screens/Auth/Auth.tsx]: HOME is bad name here

File: src/screens/Events/scenes/AddEvent/AddEvent.tsx [hunk 31]
```diff
@@ -1,18 +1,28 @@
-import React, { Component } from 'react';
+import React from 'react';
```
Comment (dimagreat) [inline src/screens/Events/scenes/AddEvent/AddEvent.tsx]: import * as React from 'react'

class extends React.Component

We don't need extra import

File: src/screens/Events/scenes/AddEvent/components/EventTypeButtons.tsx [hunk 32]
```diff
@@ -1,10 +1,20 @@
...
-  goToEditEvent = () => {
+interface ICreateEventButtonsProps {
+  option: string;
+  goToEditEvent(option: string): void;
+  title: string;
+}
+
+export class CreateEventButtons extends Component<
+  ICreateEventButtonsProps,
+  {}
```
Comment (dimagreat) [inline src/screens/Events/scenes/AddEvent/components/EventTypeButtons.tsx]: We could skip state

File: src/screens/Events/scenes/AddEvent/AddEvent.tsx (hunk 31 above)
Comment (dimagreat) [reply src/screens/Events/scenes/AddEvent/AddEvent.tsx]: Do this in all components

File: src/screens/Events/scenes/EditEvent/EditEvent.tsx [hunk 33]
```diff
@@ -31,69 +62,67 @@ export class EditEventComponent extends Component {
       },
     }));
   }
-  validate = (option, value) => {
+  public validate = (option: string, value: string) => {
```
Comment (dimagreat) [inline src/screens/Events/scenes/EditEvent/EditEvent.tsx]: this is `private` method. Here and in another components

File: src/screens/Events/scenes/Events/Events.tsx (hunk 22 above)
Comment (dimagreat) [reply src/screens/Events/scenes/Events/Events.tsx]: let's add no-console rule

File: src/screens/Events/scenes/Events/Events.tsx [hunk 34]
```diff
@@ -18,13 +37,14 @@ export class EventsComponent extends Component {
       />
     );
   };
-  keyExtractor = (item) => {
+  private keyExtractor = (item: any) => {
+    //!!!!!!!
```
Comment (dimagreat) [inline src/screens/Events/scenes/Events/Events.tsx]: what is this comment?

File: src/screens/UserPage/UserPage.tsx [hunk 35]
```diff
@@ -7,13 +7,22 @@ import {
...
-import { AUTH } from 'constants/screenCases';
-import { strings } from 'constants/strings';
+import { authService } from '../../services/authService';
+import { AUTH } from '../../constants/screenCases';
+import { strings } from '../../constants/strings';
 
-export class UserPageCompopnent extends React.Component {
+interface IPropsUserPageCompopnent {
+  logOut(): void;
+  userData: any;
```
Comment (dimagreat) [inline src/screens/UserPage/UserPage.tsx]: no-any

**Pull request merged**